### ISSUE TITLE
Reproducibility Issue

### ISSUE THREAD

**Issue by akshitac8** [sauradip/STALE#8]
Hello @sauradip For Activity using the CLIP features, I was able to these results. It would be really helpful if you could please let me know if any changes in the threshold or any other parameter is needed.

Average-mAP: 0.23484338756421028
Detection: average-mAP 23.484 mAP@0.50 40.221 mAP@0.55 36.406 mAP@0.60 33.502 mAP@0.65 30.057 mAP@0.70 26.769 mAP@0.75 22.584 mAP@0.80 18.366 mAP@0.85 14.428 mAP@0.90 8.962 mAP@0.95 3.549

**Comment by sauradip** [comment]
We will release updated code soon. This version lacks the learnable prompt part. It should improve the score after addition.

**Comment by akshitac8** [comment]
Also, can you please share model weights for 50 class split?

**Comment by sauradip** [comment]
The lcheckpoint can be found in this link [Checkpoint](https://drive.google.com/file/d/1UmgUV80XINOqKVk-uuF6JcENN6_wb_2f/view?usp=share_link)

**Comment by thanhhff** [comment]
Hi, have you updated the code with learnable prompt?
I looked at the studies about the learnable prompt, and in your model if you use that it will overfit into close-set and not generalize to open-set.

**Comment by ed-fish** [comment]
Hi - thanks for releasing your code. I am also having trouble reproducing the results in the paper for Activity-Net these are the results I get currently. 

Detection: average-mAP 23.886 mAP@0.50 40.305 mAP@0.55 36.526 mAP@0.60 33.205 mAP@0.65 31.019 mAP@0.70 27.298 mAP@0.75 23.413 mAP@0.80 19.188 mAP@0.85 14.596 mAP@0.90 9.057 mAP@0.95 4.255

**Comment by sauradip** [comment]
Hello All , Apologies for the non reproducible results. I am struggling to find time for this project. This result is 1.x mAP less than reported because of non-learnable Prompts. Actual reported value had learnable Prompts. I was following DenseCLIP repo while running my experiments but for this version I have shifted to HuggingFace CLIP variant so need to write custom wrappers for the learnable prompt. That should boost the performance to meet the reported result. I will do it soon.

**Comment by yunhanwang1105** [comment]
> 

Hi, thanks for the great work. May I ask when would the version with a learnable prompt module be released?

**Comment by ed-fish** [comment]
I have been trying to replicate your results and have found that because you use class scores and labels from another network the performance is lower bounded by the results from this network. I have even made the class names completely random and the performance is the same.  I have also implemented the learnable prompt and although the model overfits the performance again is the same since it doesn't use the class predictions from the model.

**Comment by sauradip** [comment]
Please change(lower)  the learning rate while training the prompt. This network is very sensitive to learning rate and the hyperparameters related to optimizers. Also we do not use all the scores from another network. We use 'some' scores. You can check it here : 
​
​> FIrst, We are not using UNET entirely for all the classes.
> 
> https://github.com/sauradip/STALE/blob/a574ca630a5d2658eeb47798aa94a53d3188bf07/utils/postprocess_utils.py#L138
> 
> 
> Here we only use the classes from UNET where STALE classifier performs worse. So it is a form of refinement and not replacing our classifier like [TencentYoutuResearch/ActionDetection-AFSD#4](https://github.com/TencentYoutuResearch/ActionDetection-AFSD/issues/4)
> Second, Its true CLIP is used for classification, but a thing to note why we need refinement of UNET stems from the fact that CLIP was never designed for videos. This was a first attempt to show how CLIP can be adapted. For strong generalization it is recommneded to unfreeze the backbone and use ActionCLIP pretraining instead of vanilla CLIP pretraining. We did not have the resources hence we stuck to pre-trained CLIP.
> 
> Third, This problem is very specific to ActivityNet ( check AFSD, BMN) where only 3 min videos per class does not have enough frames for classification. This refinement has been followed by TAD community for years. [wangxiang1230/SSTAP#13](https://github.com/wangxiang1230/SSTAP/issues/13) this is a semi-supervised technique where partial labels are available but still the authors follow UNET , hence this is a problem specific to ActivityNet dataset. Such problems does not arise in THUMOS dataset

**Comment by ed-fish** [comment]
I checked how many cls predictions are taken from the other network and how many are taken from STALE classifier. The STALE model predicts only ~1% of the class predictions. So to say this network performs zero shot localisation **and** classification is very misleading. Perhaps you could add a note to the paper and the repository to mention that you use these other predictions so that other people don't waste time trying to reproduce/beat the results in a true zero-shot setup.
